source/common/common/utility.cc

```diff
@@ -248,14 +249,25 @@ std::string StringUtil::escape(const std::string& source) {
 }
 
 std::string AccessLogDateTimeFormatter::fromTime(const SystemTime& time) {
-  static DateFormatter date_format("%Y-%m-%dT%H:%M:%S");
+  static DateFormatter default_date_format("%Y-%m-%dT%H:%M:%S");
 
   return fmt::format(
-      "{}.{:03d}Z", date_format.fromTime(time),
+      "{}.{:03d}Z", default_date_format.fromTime(time),
       std::chrono::duration_cast<std::chrono::milliseconds>(time.time_since_epoch()).count() %
           1000);
 }
 
+std::string AccessLogDateTimeFormatter::fromTimeWithFormat(const SystemTime& time,
+                                                           const std::string& format) {
+
+  if (!format.empty()) {
+    DateFormatter formatter(format);
```

Comment:
To avoid having to construct a DateFormatter each time, can you make the DateFormatter a member variable of StartTimeFormatter?  And then make the default date format string available as a constant somewhere, so you can construct the DateFormatter with it in the case of no args.